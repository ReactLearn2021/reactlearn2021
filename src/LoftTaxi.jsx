import './LoftTaxi.sass';
import LoginPage from "./Components/Login/LoginPage";
import React from "react";
import { MapWithAuth } from "./Components/Map/Map";
import { withAuth } from "./Components/AuthContext";
import propTypes from "prop-types";

function App(props) {
  // if (window.localStorage.getItem("view") != "map") {
    if (!props.loggedIn) {
        return (
          <main data-testid = "main">
              <LoginPage/>
          </main>
        );
    } else {
      return <main data-testid = "map-wrapper">
                  <MapWithAuth {...props} />
             </main>
    }
}

App.propTypes = {
    loggedIn : propTypes.bool,
    logIn : propTypes.func,
    logOut : propTypes.func
};

export default withAuth(App);
