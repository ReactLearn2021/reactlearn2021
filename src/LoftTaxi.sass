// body
//     overflow: hidden

@font-face
    font-family: Lora
    src: url(/assets/fonts/Lora-VariableFont_wght.ttf)

body
    margin: 0
    font-family: Lora
    -webkit-font-smoothing: antialiased
    -moz-osx-font-smoothing: grayscale

*
    outline: none

code 
    font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New', monospace

html, body, #root, section, main
    height: 100%

.left__logo-image
    height: 100%
    width: 34%
    background: #1C1A19
    display: flex
    flex-flow: row nowrap
    align-items: center
    justify-content: center

    img
        margin-right: 6%
        -webkit-user-select: none
        user-select: none

.auth__section
    background-image: url("./assets/auth-image.jpg")
    background-size: cover
    height: 100%
    width: 66%
    display: flex
    align-items: center
    justify-content: center

.authorize__block
    background: white
    padding: 46px 0
    box-shadow: 0px 0px 40px 6px rgba(0, 0, 0, 0.1)
    border-radius: 22px

.authorize__block-form
    padding: 0 102px 0 98px
    display: flex
    flex-flow: column nowrap
    justify-content: center

    h2
        text-align: center
        margin-bottom: 50px

sup
    color: firebrick

label
    position: relative
    bottom: 31px
    left: 20px
    transition: transform 200ms ease-out
    cursor: text

.loft__form-input
    width: 300px
    margin-left: 20px
    height: 35px
    border: none
    border-bottom: 1px solid black !important
    font-size: 16px
    margin-top: 20px

    &:focus + label
        transform: translateY(-31px) translateX(-34px) scale(0.8)
        transition: transform 200ms ease-out

    &:not([value=""]) + label
         transform: translateY(-31px) translateX(-34px) scale(0.8) !important

.loft__form-button
    font-size: 20px
    background-color: #D8D7D5
    color: #737373
    margin-top: 30px
    padding: 12px 15px
    border-radius: 40px
    border: none
    cursor: pointer

    &+div
        margin-top: 30px
        font-size: 17px
        
        a
            color: rgb(253, 191, 90)
            text-decoration: none

    &-filled
        background: #FDBF5A
        color: black

        &:focus
            background: #FFA842

.loft__form-anchor
    text-align: center
    width: 350px