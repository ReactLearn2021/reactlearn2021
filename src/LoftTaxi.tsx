import './LoftTaxi.sass';
import { ReactElement } from "react";
import LoginPage from "./Login/LoginPage";

function App(): ReactElement {
  return (
    <main>
        <LoginPage/>
    </main>
  );
}

export default App;
