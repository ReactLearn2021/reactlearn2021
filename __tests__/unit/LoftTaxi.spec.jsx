/**
 * @jest-environment jsdom
 */

import React from "react";
import { render } from "@testing-library/react";
import LoftTaxi from '../../src/LoftTaxi';
import renderer from "react-test-renderer";

jest.mock('mapbox-gl/dist/mapbox-gl', () => ({
    Map: jest.fn(() => ({
      remove: jest.fn(),
    })),
}));

 
test("it renders correctly", () => {
    const loftTree = renderer.create(<LoftTaxi />).toJSON(),
          { getByTestId } = render(<LoftTaxi loggedIn={true} />);
    expect(loftTree).toMatchSnapshot();
    expect(getByTestId("map-wrapper").firstChild).toBeTruthy(); // потому что loggedIn === true
});