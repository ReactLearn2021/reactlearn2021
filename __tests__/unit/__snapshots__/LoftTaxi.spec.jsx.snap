// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<main
  data-testid="main"
>
  <section
    id="loginPage"
  >
    <div
      className="left__logo-image"
    >
      <img
        alt="Logotip was not loaded"
        data-testid="logo"
        src={
          Object {
            "process": [Function],
          }
        }
      />
    </div>
    <div
      className="auth__section"
    >
      <div
        className="authorize__block"
      >
        <div
          className="authorize__block-form"
        >
          <h2>
            Войти
          </h2>
          <input
            className="loft__form-input"
            data-testid="login"
            id="login"
            name="login"
            onChange={[Function]}
            onFocus={[Function]}
            type="text"
            value=""
          />
           
          <label
            htmlFor="login"
          >
            Имя пользователя 
            <sup>
              ❉
            </sup>
          </label>
          <input
            className="loft__form-input"
            data-testid="password"
            id="password"
            name="password"
            onChange={[Function]}
            onFocus={[Function]}
            type="password"
            value=""
          />
          <label
            htmlFor="password"
          >
            Пароль 
            <sup>
              ❉
            </sup>
          </label>
          <button
            className="loft__form-button"
            data-testid="login-button"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            Войти
          </button>
          <div
            className="loft__form-anchor"
          >
            <span>
              Новый пользователь? 
              <a
                onClick={[Function]}
              >
                Зарегистрироваться
              </a>
            </span>
          </div>
        </div>
      </div>
    </div>
  </section>
</main>
`;
